internal/docstore/firedocstore/fs.go

```diff
@@ -634,23 +648,37 @@ func toServiceFieldPathComponent(key string) string {
 // revisionPrecondition returns a Firestore precondition that asserts that the stored document's
 // revision matches the revision of doc.
 func revisionPrecondition(doc driver.Document) (*pb.Precondition, error) {
+	rev, err := revisionTimestamp(doc)
```

Comment:
Why create these two functions? The 3 names look similar and may cause some confusion and it seems you can use the `revisionPrecondition` in the above scenarios. At least one of the three can be saved.

Reply:
I use preconditionFromTimestamp in newUpdateWrites and revisionTimestamp in updateWrites.